src/core/server/config/schema/types/__tests__/string_type.test.ts

```diff
@@ -53,6 +53,42 @@ describe('#maxLength', () => {
   });
 });
 
+describe('#hostname', () => {
+  test('returns value for valid hostname as per RFC1123', () => {
+    const hostNameSchema = schema.string({ hostname: true });
+
+    expect(hostNameSchema.validate('www.example.com')).toBe('www.example.com');
+    expect(hostNameSchema.validate('3domain.local')).toBe('3domain.local');
+    expect(hostNameSchema.validate('hostname')).toBe('hostname');
+    expect(hostNameSchema.validate('2387628')).toBe('2387628');
+    expect(hostNameSchema.validate('::1')).toBe('::1');
+    expect(hostNameSchema.validate('0:0:0:0:0:0:0:1')).toBe('0:0:0:0:0:0:0:1');
+    expect(hostNameSchema.validate('xn----ascii-7gg5ei7b1i.xn--90a3a')).toBe(
+      'xn----ascii-7gg5ei7b1i.xn--90a3a'
+    );
+
+    const hostNameWithMaxAllowedLength = Array(255)
```

Comment:
`'a'.repeat(255)` would do the trick too 😄

Reply:
Damn, much better, thanks! :heavy_check_mark: